### PR TITLE
Add support for SARIF format with --sarif

### PR DESCRIPTION
**Pull request by underyx**
Closes https://github.com/returntocorp/enterprise/issues/140

### CODE REVIEW

File: semgrep/semgrep/output.py [hunk 1]
```diff
@@ -105,15 +109,48 @@ def build_output_json(rule_matches: List[RuleMatch], semgrep_errors: List[Any])
...
         return build_output_json(rule_matches, semgrep_errors)
-    else:
+    elif output_format == "sarif":
+        return build_sarif_output(rule_matches, rules, semgrep_errors)
+    elif output_format == "normal":
         return "\n".join(
             build_normal_output(rule_matches, semgrep_errors, color_output)
         )
+    else:
+        raise RuntimeError(f"Not sure how to format as '{output_format}'")
```
Comment (underyx) [inline semgrep/semgrep/output.py]: This is to satisfy the type hints. Should never happen as we control the `output_format` value in our code.

**Review by DrewDennison — APPROVED** [review]
👏

**Comment by ievans** [comment]
@underyx can you also update https://github.com/returntocorp/semgrep/blob/develop/docs/output.md ?

and could we maybe note --json and --sarif flags in the root README?

**Comment by dlukeomalley** [comment]
Thoughts on adopting Bento's `--output=` flag since we'll have many different output formats over the lifespan of semgrep?

Currently / moving forward:

* json
* sarif
* default (whatever we have now)
* clippy
* histo

With different output flags, is it possible to output in more than one format at a time, and if not, which flags take precedence?

**Comment by ievans** [comment]
Personally I really disliked the ergonomics of the old flag and would like to keep `--json` as a top-level flag. Less opinionated about sarif.

**Comment by dlukeomalley** [comment]
Good feedback. Are you regularly typing out `--json` or `--output=json`? Figured it was more one-off during scripting time.

**Review by brendongo — APPROVED** [review]

**Comment by brendongo** [comment]
Add to release notes?

**Comment by underyx** [comment]
Made all the changes except for `--output`. I think the ideal design would be:

- either set `--output="normal | json | sarif | histo | clippy"`
- or for globally known formats, use shorthand with their names: `--json`, `--sarif`, `--xml`
- internal names such as `clippy` and `histo` will never get a shorthand flag

Since this PR's implementation is a subset of the above proposal, I've left it for now with only the second point implemented.

**Review by brendongo — APPROVED** [review]

**Pull request merged**